homeassistant/components/modbus/base_platform.py

```diff
@@ -74,8 +76,34 @@
 class BasePlatform(Entity):
     """Base for readonly platforms."""
 
-    def __init__(self, hub: ModbusHub, entry: dict[str, Any]) -> None:
+    def __init__(
+        self, hass: HomeAssistant, hub: ModbusHub, entry: dict[str, Any]
+    ) -> None:
         """Initialize the Modbus binary sensor."""
+
+        if CONF_LAZY_ERROR in entry:
+            async_create_issue(
```

Comment:
Instantiating an object shouldn't have side effects. I suggest moving this out to something that checks the `entry`.

Reply:
Good point! I will make a follow up PR a bit later and request you as reviewer.

Reply:
I think we could even move this to the config validation as we can now get access to `hass` there.

https://github.com/home-assistant/core/blob/88a4b74d4befead68ea11a7cf0723e2aaa4c1bf8/homeassistant/core.py#L235

Reply:
wow, thanks !

I was not aware of that option, with that I can move a number of checks out of the production code.